### ISSUE TITLE
Bug: PHPStan is broken because of service('auth')->routes($routes);

### ISSUE THREAD

**Issue by jozefrebjak** [codeigniter4/shield#610]
### PHP Version

8.1.14

### CodeIgniter4 Version

4.3.1

### Shield Version

dev-devlop

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

MariaDB 10.2

### Did you customize Shield?

No

### What happened?

PHPStan is broken.

### Steps to Reproduce

Install appstarter

```sh
composer create-project codeigniter4/appstarter shield-phpstan-issue
```

Go to a directory

```sh
cd shield-phpstan-issue
```

Install shield

```sh
composer require codeigniter4/shield:dev-develop
```

Setup Shield

```sh
php spark shield:setup
```

> Here we can skip a database migration.

Install PHPStan

```sh
composer require --dev phpstan/phpstan
```

Create configuration for a PHPStan like:

```
parameters:
    tmpDir: build/phpstan
    level: 5
    paths:
        - app
        - tests
    bootstrapFiles:
        - vendor/codeigniter4/framework/system/Test/bootstrap.php
    excludePaths:
        - **/Config/Routes.php
        - **/Views/*
    universalObjectCratesClasses:
        - CodeIgniter\Entity
        - CodeIgniter\Entity\Entity
        - Faker\Generator
        - Faker\Factory
    scanDirectories:
        - app/Helpers
        - vendor/codeigniter4/framework/system/Helpers
        - vendor/codeigniter4/settings/src/Helpers
        - vendor/codeigniter4/shield/src/Helpers
    dynamicConstantNames:
        - APP_NAMESPACE
        - CI_DEBUG
        - ENVIRONMENT
```

Try to analyse with PHPstan

```sh
vendor/bin/phpstan analyse
```

And there is an error:

```sh
Note: Using configuration file /Users/ME/CodeIgniter4/shield-phpstan-issue/phpstan.neon.dist.
Error thrown in /Users/ME/CodeIgniter4/shield-phpstan-issue/app/Config/Routes.php on line 34 while loading bootstrap file /Users/ME/CodeIgniter4/shield-phpstan-issue/vendor/codeigniter4/framework/system/Test/bootstrap.php: Call to a member function routes() on null
```

This error is there also in Docker environment.

### Expected Output

PHPStan should work.

### Anything else?

This error is there from CodeIgniter4 version 4.3.0. I'm opening issue there because Shield setup and docs are using 

```php
service('auth')->routes($routes);
```

to propagate routes.

**Comment by kenjis** [comment]
Thank you.
I confirmed this issue.

**Comment by kenjis** [comment]
@jozefrebjak Try https://github.com/codeigniter4/CodeIgniter4/pull/7193

**Comment by kenjis** [comment]
@jozefrebjak Thank you for testing!
The PR was merged, so this issue will be fixed in CI v4.3.2.
